localstack/services/cloudformation/models/kinesis.py

```diff
@@ -52,11 +56,27 @@ def get_deploy_templates():
         def get_delete_params(params, **kwargs):
             return {"StreamName": params["Name"], "EnforceConsumerDeletion": True}
 
+        def _store_arn_and_wait_until_ready(result, resource_id, resources, resource_type):
+            client = aws_stack.connect_to_service("kinesis")
+            stream_name = resources[resource_id]["Properties"]["Name"]
+
+            description = client.describe_stream(StreamName=stream_name)
+
+            resources[resource_id]["PhysicalResourceId"] = description["StreamDescription"][
+                "StreamARN"
+            ]
+            latest_status = description["StreamDescription"]["StreamStatus"]
+            while latest_status == "CREATING":
+                time.sleep(0.5)
+                description = client.describe_stream(StreamName=stream_name)
+                latest_status = description["StreamDescription"]["StreamStatus"]
```

Comment:
For this we usually use waiters (in this case you'd wait for the 'ACTIVE' State).

Unfortunately though this doesn't fit with our current template deployer implementation since the resources aren't being created concurrently right now, so this would block all other resources and in the worst case might lead to a deadlock.

Given the current model, what you should do here is check the status in `fetch_state` and if it's not `ACTIVE`, just return None. This will signal the template_deployer that the deployment hasn't completed yet.